### PR TITLE
CI: fix macos unbreaking python

### PR DESCRIPTION
**Pull request by dcbaker**
The install location seems to have moved, which breaks our job to unbreak things.

### CODE REVIEW

**Review by tristan957 — APPROVED** [review]

File: mesonbuild/dependencies/boost.py [hunk 1]
```diff
@@ -688,8 +689,12 @@ def detect_roots(self) -> None:
             tmp: T.List[Path] = []
 
             # Add some default system paths
+            if m.is_darwin():
+                tmp.extend([
+                    Path('/opt/homebrew/'),        # for Apple Silicon MacOS
+                    Path('/usr/local/opt/boost'),  # for Intel Silicon MacOS
+                ])
             tmp += [Path('/opt/local')]
-            tmp += [Path('/usr/local/opt/boost')]
```
Comment (eli-schwartz) [inline mesonbuild/dependencies/boost.py:692]: This path is specific to Apple? Good to know...

File: .github/workflows/macos.yml [hunk 2]
```diff
@@ -102,8 +102,8 @@ jobs:
         python3 -m pip install --upgrade pip
         python3 -m pip install cython coverage
     - env:
-        CPPFLAGS: "-I/opt/homebrew/include"
-        LDFLAGS: "-L/opt/homebrew/lib"
+        CPPFLAGS: "-I/opt/homebrew/opt/include"
+        LDFLAGS: "-L/opt/homebrew/opt/lib"
         MESON_ARGS: --unity=${{ matrix.unity }}
```
Comment (eli-schwartz) [inline .github/workflows/macos.yml:107]: Is this change in the right commit? Why isn't it done in unittests-appleclang? Why is homebrew using a double /opt anyway...

File: mesonbuild/dependencies/boost.py (hunk 1 above)
Comment (dcbaker) [reply mesonbuild/dependencies/boost.py:692]: Yeah, homebrew on Intel puts its stuff there. I don't think it's a place we should necessary search automatically. I'm a little skeptical of some of the paths in there in general. Handling homebrew seems reasonable as it's pretty much de facto for Apple

File: .github/workflows/macos.yml (hunk 2 above)
Comment (dcbaker) [reply .github/workflows/macos.yml:107]: grrrrrrrrrrr. that rebased *back in*!

**Pull request merged**